### PR TITLE
ci: add replica-healthcheck to image publish workflow

### PR DESCRIPTION
**Pull request by annieke**
**Description**
Adding replica-healthcheck to be published as a docker image since some providers are interested in using it!

**Additional context**
I didn't add a script for this to be run along the other services and also didn't add it to docker-compose, happy to do so if we think this belongs in the dafault stack though!

**Metadata**
- Fixes OP-1207

### CODE REVIEW

File: ops/docker/Dockerfile.replica-healthcheck [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+COPY --from=builder /optimism/packages/contracts/artifacts ./packages/contracts/artifacts
+COPY --from=builder /optimism/packages/contracts/artifacts-ovm ./packages/contracts/artifacts-ovm
+
+# copy the service
+WORKDIR /opt/optimism/packages/replica-healthcheck
+COPY --from=builder /optimism/packages/replica-healthcheck/dist ./dist
+COPY --from=builder /optimism/packages/replica-healthcheck/package.json .
+COPY --from=builder /optimism/packages/replica-healthcheck/node_modules ./node_modules
+
+ENTRYPOINT ["node", "dist/src/exec/run-healthcheck-server.js"]
```
Comment (optimisticben) [inline ops/docker/Dockerfile.replica-healthcheck:32]: `ENTRYPOINT ["node", "dist/src/exec/run-healthcheck-server.js"]`
should be
`ENTRYPOINT ["node", "dist/exec/run-healthcheck-server.js"]` ?

**Review by optimisticben — CHANGES_REQUESTED** [review]

**Comment by tynes** [comment]
Make sure to try building and running the docker image locally @annieke and let us know if you got it working

File: ops/docker/Dockerfile.replica-healthcheck (hunk 1 above)
Comment (annieke) [reply ops/docker/Dockerfile.replica-healthcheck]: good catch! fixed and tested locally that this runs from the correct entrypoint

**Comment by annieke** [comment]
@tynes tested and got the images running correctly locally; also added it to the canary publish so can test in ci as well! skipped doing that here since it'll publish all images

File: ops/docker/Dockerfile.replica-healthcheck [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM ethereumoptimism/builder AS builder
+
+FROM node:14-alpine
+
+RUN apk add --no-cache curl bash jq
```
Comment (tynes) [inline ops/docker/Dockerfile.replica-healthcheck:5]: I don't think `jq` is technically required here, is it?

**Comment by tynes** [comment]
> @tynes tested and got the images running correctly locally; also added it to the canary publish so can test in ci as well! skipped doing that here since it'll publish all images

It would definitely be ideal to post exactly how you built/tested the image locally to make it super easy for another team member to reproduce

**Comment by annieke** [comment]
@tynes tested this full on with the env file instead of just the entrypoint and found a bug 😅 should work as expected now!

1. ran `docker compose --build` in `ops` directory to have updated `builder` image
2. ran `docker build - < docker/Dockerfile.replica-healthcheck` also in `ops` after `builder` was done building
3. `docker run --env-file .env <image-hash>` with the hash from the previous step and the following `.env`:
```
REPLICA_HEALTHCHECK__ETH_NETWORK=mainnet
REPLICA_HEALTHCHECK__ETH_NETWORK_RPC_PROVIDER=https://mainnet-sequencer.optimism.io
REPLICA_HEALTHCHECK__ETH_REPLICA_RPC_PROVIDER=https://mainnet.optimism.io
REPLICA_HEALTHCHECK__L2GETH_IMAGE_TAG=0.4.8
```
(used our public replica)

4. logs
```
{"level":30,"time":1629161878859,"msg":"Listening on port 3000"}
{"level":30,"time":1629161879385,"blockNumber":670294,"msg":"State roots matching"}
{"level":30,"time":1629161879883,"sequencerHeight":670294,"replicaHeight":670294,"heightDifference":0,"msg":"Syncing from sequencer"}
{"level":30,"time":1629161879883,"msg":"Replica caught up with sequencer, waiting for next block"}
```

**Comment by optimisticben** [comment]
> REPLICA_HEALTHCHECK__L2GETH_IMAGE_TAG

I don't think this should be a required env (not sure how it would be used in this context).

Current deployments don't need it 😁  https://github.com/ethereum-optimism/k8s/blob/master/kustomize/replica-healthcheck/kovan-quicknode/replica-healthcheck.env

**Comment by annieke** [comment]
@optimisticben we use it as a metrics label here:
https://github.com/ethereum-optimism/optimism/blob/796dbda597bf249cf31dfe4feb026c9968e26aaf/packages/replica-healthcheck/src/healthcheck-server.ts#L44-L49
is that populated for our deployments?

**Comment by optimisticben** [comment]
> is that populated for our deployments?

No, they still use the old version from the container in my namespace.

It's not part of this PR, just new to me while testing 😄

**Review by optimisticben — APPROVED** [review]
LGTM

**Review by snario — APPROVED** [review]

**Pull request merged**